packages/@aws-cdk/aws-ec2/lib/client-vpn.ts

```diff
@@ -0,0 +1,212 @@
+import cdk = require('@aws-cdk/core');
+import {
+  CfnClientVpnAuthorizationRule,
+  CfnClientVpnEndpoint,
+  CfnClientVpnRoute,
+  CfnClientVpnTargetNetworkAssociation
+} from './ec2.generated';
+import { ISubnet } from './vpc';
+
+export interface IClientVpnEndpoint extends cdk.IResource {
+  /**
+   * The id of the Client VPN Endpoint
+   */
+  readonly clientVpnEndpointId: string;
+}
+
+// TODO helper class if activeDirectory and mutualAuthentication mutually exclusive?
+export interface IClientAuthenticationRequest {
+  readonly activeDirectoryId?: string;
+  readonly mutualAuthenticationClientRootCertificateChainArn?: string;
+  readonly type: ClientRequestAuthenticationType;
+}
+
+export interface IConnectionLogOptions {
+  // TODO replace with cloudwatch object?
+  readonly cloudwatchLogGroup?: string;
+  readonly cloudwatchLogStream?: string;
+  readonly enabled: boolean;
+}
+
+export interface ITagSpecificationTag {
+  readonly key: string;
+  readonly value: string;
+}
+
+export interface ITagSpecification {
+  readonly resourceType: TagSpecificationResourceType;
+  readonly tags: ITagSpecificationTag[];
+}
+
+export interface ClientVpnEndpointProps {
+  readonly authenticationOptions: IClientAuthenticationRequest[];
+  readonly clientCidrBlock: string;
+  readonly connectionLogOptions: IConnectionLogOptions;
+  // TODO replace with serverCertificate object?
```

Comment:
Unfortunately, I don't think it's possible:
```
Dependency cycles detected, you should fix these!
aws-cdk/aws-certificatemanager -> @aws-cdk/aws-ec2 -> @aws-cdk/aws-lambda -> @aws-cdk/aws-certificatemanager
```

Any way to work around this?

Reply:
I'm sorry, where are these dependencies coming from?

On my machine, I'm not seeing any dependency from EC2 to either Lambda or CertificateManager.

Reply:
I was trying to add `@aws-cdk/aws-certificatemanager` to EC2 to get access to `ICertificate`s. I haven't committed it because of the previous error

Reply:
Is it possible to use both IAM and ACM certificates here? I've been thinking about this for a while. A potential solution might be to declare an `iam.ICertificate` (or `iam.ICertificateSource`), and have ACM's `ICertificate` extend that. Since everything depends on IAM already, we wouldn't have any dependency issues.

Reply:
I don't think it's possible to use IAM certificates in this case, unless I'm missing something:

[Getting started](https://docs.aws.amazon.com/vpn/latest/clientvpn-admin/cvpn-getting-started.html):

> The server certificate must be provisioned in AWS Certificate Manager (ACM)

[Client Authentication and Authorization](https://docs.aws.amazon.com/vpn/latest/clientvpn-admin/authentication-authorization.html):

> The server and client certificates must be uploaded to AWS Certificate Manager (ACM).